src/cloud.c

```diff
@@ -329,37 +341,27 @@ int cloud_unregister_device(const char *id)
  */
 int cloud_auth_device(const char *id, const char *token)
 {
-	amqp_bytes_t queue_cloud;
 	json_object *jobj_auth;
 	const char *json_str;
 	int result;
 
-	queue_cloud = mq_declare_new_queue(MQ_QUEUE_FOG_IN);
-	if (queue_cloud.bytes == NULL) {
-		l_error("Error on declare a new queue");
-		return -1;
-	}
-
 	jobj_auth = parser_auth_json_create(id, token);
-	if (!jobj_auth) {
-		amqp_bytes_free(queue_cloud);
+	if (!jobj_auth)
 		return KNOT_ERR_CLOUD_FAILURE;
-	}
+
 	json_str = json_object_to_json_string(jobj_auth);
 
 	headers[0].value.value.bytes = amqp_cstring_bytes(user_auth_token);
 
-	result = mq_publish_persistent_message(queue_cloud,
-					       MQ_EXCHANGE_FOG_IN,
+	result = mq_publish_direct_message_rpc(MQ_EXCHANGE_DEVICE,
 					       MQ_CMD_DEVICE_AUTH,
-					       headers, 1,
-					       0, // Set no expiration time
+					       headers, 1, 0,
```

Comment:
Add a comment informing whats is 1 and 0 or why it should be 1 and why it should be 0.

Reply:
I put a disclaimer informing about the AMQP settings of the sent message